qoqo_qiskit/src/qoqo_qiskit/backend/backend.py

```diff
@@ -48,40 +54,95 @@ def __init__(
         self.memory = memory
         self.compilation = compilation
 
-    def run_circuit(
+    # Internal _run_circuit method, for async support
+    def _run_circuit(self, circuit: Circuit) -> Tuple[
+        Job,
+        str,
+        Dict[str, int],
+        Dict[str, List[List[bool]]],
+        Dict[str, List[List[float]]],
+        Dict[str, List[List[complex]]],
+    ]:
+        if not isinstance(circuit, Circuit):
+            raise TypeError("The input is not a valid Qoqo Circuit instance.")
+
+        # Initial register setup
+        (
+            clas_regs_sizes,
+            output_bit_register_dict,
+            output_float_register_dict,
+            output_complex_register_dict,
+        ) = self._setup_registers(circuit)
```

Comment:
I think that overall in this function, the `_...` functions are very verbosely named, so you could remove the comments